plugins/performance-lab/includes/site-health/far-future-headers/helper.php

```diff
@@ -0,0 +1,320 @@
+			$conditional_pass = perflab_ffh_try_conditional_request( $asset, $headers );
+			$final_status     = 'recommended';
+			if ( ! $conditional_pass ) {
+				$fail_details[] = array(
+					'filename' => $filename,
+					'reason'   => __( 'No far-future headers and no conditional caching', 'performance-lab' ),
+				);
+			} else {
+				$fail_details[] = array(
+					'filename' => $filename,
+					'reason'   => __( 'No far-future headers but conditionally cached', 'performance-lab' ),
+				);
+			}
+		} else {
+			// If there's a max-age or expires but below threshold, we skip conditional.
+			$final_status   = 'recommended';
+			$fail_details[] = array(
+				'filename' => $filename,
+				'reason'   => $check['reason'],
+			);
+		}
+	}
+
+	return array(
+		'final_status' => $final_status,
+		'details'      => $fail_details,
+	);
+}
+
+/**
+ * Checks if far-future expiration headers are enabled.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @param WpOrg\Requests\Utility\CaseInsensitiveDictionary|array<string, string|array<string>> $headers Response headers.
+ * @return array{passed: bool, reason: string}|false Detailed result. If passed=false, reason explains why it failed and false if no headers found.
+ */
+function perflab_ffh_check_headers( $headers ) {
+	/**
+	 * Filters the threshold for far-future headers.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int $threshold Threshold in seconds.
+	 */
+	$threshold = apply_filters( 'perflab_far_future_headers_threshold', YEAR_IN_SECONDS );
+
+	$cache_control = $headers['cache-control'] ?? '';
+	$expires       = $headers['expires'] ?? '';
+
+	// Check Cache-Control header for max-age.
+	$max_age = 0;
+	if ( '' !== $cache_control ) {
+		// There can be multiple cache-control headers, we only care about max-age.
+		foreach ( (array) $cache_control as $control ) {
+			if ( 1 === preg_match( '/max-age\s*=\s*(\d+)/', $control, $matches ) ) {
+				$max_age = (int) $matches[1];
+				break;
+			}
+		}
+	}
+
+	// If max-age meets or exceeds the threshold, we consider it good.
+	if ( $max_age >= $threshold ) {
+		return array(
+			'passed' => true,
+			'reason' => '',
+		);
+	}
+
+	// If max-age is too low or not present, check Expires.
+	if ( is_string( $expires ) && '' !== $expires ) {
+		$expires_time   = strtotime( $expires );
+		$remaining_time = is_int( $expires_time ) ? $expires_time - time() : 0;
+		if ( $remaining_time >= $threshold ) {
+			// Good - Expires far in the future.
+			return array(
+				'passed' => true,
+				'reason' => '',
+			);
+		}
+
+		// Expires header exists but not far enough in the future.
+		if ( $max_age > 0 ) {
+			return array(
+				'passed' => false,
+				'reason' => sprintf(
+					/* translators: 1: actual max-age value in seconds, 2: threshold in seconds */
+					__( 'max-age below threshold (actual: %1$s seconds, threshold: %2$s seconds)', 'performance-lab' ),
+					number_format_i18n( $max_age ),
+					number_format_i18n( $threshold )
+				),
+			);
```

Comment:
Looks like a test is missing for the `max-age` being present but it not being high enough.